lua/impatient/cachepack.lua

```diff
@@ -0,0 +1,90 @@
+local ffi = require('ffi')
+
+local M = {}
+
+-- using double for packing/unpacking numbers has no conversion overhead
+local c_double = ffi.typeof("double[1]")
+local sizeof_c_double = ffi.sizeof("double")
+
+local OutputBuffer = {}
+
+function OutputBuffer.create() return {} end
+
+function OutputBuffer.write_number(buf, num)
+  buf[#buf+1] = ffi.string(c_double(num), sizeof_c_double)
+end
+
+function OutputBuffer.write_string(buf, str)
+  OutputBuffer.write_number(buf, #str)
+  buf[#buf+1] = str
+end
+
+function OutputBuffer.to_string(buf)
```

Comment:
Personally, I find this function and `OutputBuffer.create()` make the code a little harder to read. It would read better to me if they were inlined.